### ISSUE TITLE
Unexpected suggestion triggered by running script using Get-Command

### ISSUE THREAD

**Issue by sba923** [PowerShell/PowerShell#24712]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Running https://gist.github.com/sba923/d5406e7fded598b9c1131ddf9f099564#file-update-seleniumdriver-ps1 (which runs https://gist.github.com/sba923/d0d5ad16f2b12d7785adf830b0395dc2#file-whence-ps1 which in turn runs https://gist.github.com/sba923/d0d5ad16f2b12d7785adf830b0395dc2#file-whatis-ps1) yields the following when PowerShell returns to the prompt:

````
Suggestion [3,General]: The command "chromedriver.exe" was not found, but does exist in the current location.
PowerShell does not load commands from the current location by default (see ''Get-Help about_Command_Precedence'').

If you trust this command, run the following command instead:

Suggestion [4,General]: The most similar commands are:
````

PowerShell 7.4.6 issues a slightly different suggestion:

````
Suggestion [3,General]: The command "chromedriver.exe" was not found, but does exist in the current location.
PowerShell does not load commands from the current location by default (see ''Get-Help about_Command_Precedence'').

If you trust this command, run the following command instead:
````

### Expected behavior

```console
No suggestion should be emitted
```

### Actual behavior

```console
Suggestion [3,General]: The command "chromedriver.exe" was not found, but does exist in the current location.
PowerShell does not load commands from the current location by default (see ''Get-Help about_Command_Precedence'').

If you trust this command, run the following command instead:

Suggestion [4,General]: The most similar commands are:
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.5.0-rc.1
PSEdition                      Core
GitCommitId                    7.5.0-rc.1
OS                             Microsoft Windows 10.0.26100
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by sba923** [comment]
@SteveL-MSFT could that be a regression to the fix for https://github.com/PowerShell/PowerShell/issues/19862?

**Comment by 237dmitry** [comment]
> Steps to reproduce

Are you serious?

Add current location to the `$env:PATH`

```powershell
$env:PATH += ";$pwd"
chromedriver.exe
```

**Comment by sba923** [comment]
> > Steps to reproduce
> 
> Are you serious?
> 
> Add current location to the `$env:PATH`
> 
> ```powershell
> $env:PATH += ";$pwd"
> chromedriver.exe
> ```

You clearly have misunderstood what the issue is about and/or the script's code.

During the execution of the script, the current working directory is typically **not** where `chromedriver.exe` resides. The code needs to determine where the file `chromedriver.exe` that is on the `PATH` (if there is one) resides, so that it can be updated _there_ if a newer version is available.

The core issue lies in the fact a "suggestion" should never (IMVHO) be emitted by PowerShell when run non-interactively.

**Comment by SeeminglyScience** [comment]
@237dmitry Even if you were correct that would be an unacceptable way to respond. Please review the [Code of Conduct](https://opensource.microsoft.com/codeofconduct/).

**Comment by SeeminglyScience** [comment]
I'm referring to the part where you prefaced your suggestion with "Are you serious?".

**Comment by 237dmitry** [comment]
So what? I just expressed surprise at such reproduction steps. Very specific scripts and in another repo.

If for English-speaking users this looks offensive, then I apologize, in my opinion it is a completely normal question.

**Comment by jhoneill** [comment]
I'm not sure that readers will work through all the stuff linked to in order to see what the problem is. 

It looks like "I don't want a message - this is in the current directory, but that isn't on the path" 

NOW... there are two or three entirely independent things here
1. Should the "you can just cd to the executable location and run _name_" suggestion appear when _name_ is not entered at the command line. (And if _name_ is not on the path what error should running that in a script / function produce)
2. Is preventing adding . to the path really the good idea that those who changed it thought. 
3. **What I think is the actual question**  - is this behaviour of Get-Command by design and if so is the design right

![image](https://github.com/user-attachments/assets/cff7aa93-344a-49be-b580-1f8e6aea6900)

i.e. If you are in a directory where the executable is found AND that directory is not on the path, then Get-Command generates the suggestion  - and a bit like `-whatif`  messages they can't be captured

**Comment by sba923** [comment]
Indeed, you're perfectly right, I must confess I didn't analyze the issue like I had done for https://github.com/PowerShell/PowerShell/issues/19862. Time to fix that.

My key point is that PowerShell should not emit suggestions unless commands are entered interactively.

It seems unclear as to what is causing this rule to be broken (again, as this feels like a regression).

`Get-Command` seems to emit that suggestion whenever the value of the `-Name` parameter cannot be found on the `PATH` _but_ it's present in the current directory.

Probably the code for `Get-Command` is invoked when a command line is evaluated prior to execution, which **is** OK when this happens interactively.

There are other situations where non-interactive operations (incorrectly, IMVHO) result in the suggestion to be emitted:

![image](https://github.com/user-attachments/assets/98dbf930-4bc3-41f6-bba9-5e2002413d3c)

I don't know whether PowerShell would be able to detect whether code is executed from a script and would disable suggestions in that case.

**Comment by jhoneill** [comment]
Here are some more cases . 
![image](https://github.com/user-attachments/assets/c99e4b76-6307-4d5d-8fdb-738df670a03f)

No, I don't know if `start-process` should work differently to `&`  

I think the ask to detect running in a script and change behaviour is do-able but the wrong answer;  as a rule things should do the same thing at the prompt as they do in a script (although there can be exceptions) 

Personally I have more of a problem that -ErrorAction doesn't supress this message, and it can't be redirected or caught in any way. It should be possible to set some kind of preference variable to make it go away. If people want to turn it off globally, or just in their script - those should be possible.

**Comment by sba923** [comment]
I think there should be an exception to that rule: suggestions are meant to inform the interactive user only. Emitting those while a script is running, with potentially no one looking at the output (think about scheduled tasks, for instance) doesn't make much sense to me. They are **not** the same as error information.

BTW, there's another issue about those suggestions: they seem to be incomplete, as they end with "you should do this:" without telling the user what "this" should be...

**Comment by sba923** [comment]
Consider `bash`'s behavior on Ubuntu 24:

![image](https://github.com/user-attachments/assets/efcf9823-0e7a-4a0d-8e37-c719aa4203e5)

The suggestion to the interactive user is **not** emitted when the same command is evaluated during script execution.

**Comment by mklement0** [comment]
All good points, @sba923.

As for the following incidental aspect:

> there's another issue about those suggestions: they seem to be incomplete

That it is indeed unfortunate; see:

* #20916

**Comment by sba923** [comment]
This is still present in 7.5.4...
